### PR TITLE
Added some features

### PR DESCRIPTION
**Pull request by evgeny-izutov**
Adds features:
* Classification pipeline
* InfoDrop
* Mutual learning
* Mixing loss (some kind of AugMix)
* Contrastive learning with InfoNCE loss

FYI @sovrasov

### CODE REVIEW

File: scripts/default_config.py [hunk 1]
```diff
@@ -15,19 +15,25 @@ def get_default_config():
...
+    cfg.model.IN_conv1 = False
     cfg.model.extra_blocks = False
     cfg.model.lct_gate = False
     cfg.model.fpn = CN()
     cfg.model.fpn.enable = True
     cfg.model.fpn.dim = 256
     cfg.model.fpn.process = 'concatenation'
+    cfg.model.classification = False
+    cfg.model.contrastive = False
+    cfg.model.pool_method = 'avg'
```
Comment (sovrasov) [inline scripts/default_config.py]: This parameter duplicates `cfg.model.pooling_type`

File: torchreid/losses/metric.py [hunk 2]
```diff
@@ -206,7 +206,7 @@ def __init__(self, num_classes, embed_size,
             self.losses_map['center'] = self.total_losses_num
             self.total_losses_num += 1
 
-            self.centers_push_loss = CentersPush(margin=0.1)
+            self.centers_push_loss = CentersPush(margin=0.9)
```
Comment (sovrasov) [inline torchreid/losses/metric.py]: Is this parameter important or hardcoded value is ok in most cases?

File: torchreid/engine/image/contrastive.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from torchreid.engine import Engine
+from torchreid.losses import get_regularizer, InfoNCELoss
+
+
+class ImageContrastiveEngine(Engine):
+    r"""AM-Softmax-loss engine for image-reid.
```
Comment (sovrasov) [inline torchreid/engine/image/contrastive.py]: InfoNCELoss loss engine ...

File: scripts/default_config.py (hunk 1 above)
Comment (evgeny-izutov) [reply scripts/default_config.py]: Removed the duplication.

File: torchreid/losses/metric.py (hunk 2 above)
Comment (evgeny-izutov) [reply torchreid/losses/metric.py]: Moved parameters definition into `default_config.py`.

File: torchreid/engine/image/contrastive.py (hunk 3 above)
Comment (evgeny-izutov) [reply torchreid/engine/image/contrastive.py]: Fixed.

**Review by sovrasov — APPROVED** [review]
